pkg/sql/importer/import_processor.go

```diff
@@ -219,11 +234,37 @@ func (idp *readImportDataProcessor) Next() (rowenc.EncDatumRow, *execinfrapb.Pro
 	if err != nil {
 		return nil, idp.DrainHelper()
 	}
-
-	return rowenc.EncDatumRow{
+	// When using distributed merge, the processor will emit the SST's and their
+	// start and end keys.
+	var fileDataums rowenc.EncDatumRow
+	if len(idp.files) > 0 {
+		fileNames := tree.NewDArray(types.String)
+		startKeys := tree.NewDArray(types.Bytes)
+		endKeys := tree.NewDArray(types.Bytes)
+		for _, file := range idp.files {
+			if err := fileNames.Append(tree.NewDString(file.FileName)); err != nil {
+				idp.MoveToDraining(err)
+				return nil, idp.DrainHelper()
+			}
+			if err := startKeys.Append(tree.NewDBytes(tree.DBytes(file.Span.Key))); err != nil {
+				idp.MoveToDraining(err)
+				return nil, idp.DrainHelper()
+			}
+			if err := endKeys.Append(tree.NewDBytes(tree.DBytes(file.Span.EndKey))); err != nil {
+				idp.MoveToDraining(err)
+				return nil, idp.DrainHelper()
+			}
+		}
+		fileDataums = rowenc.EncDatumRow{
+			rowenc.DatumToEncDatum(types.StringArray, fileNames),
```

Comment:
What do you think of returning a protobuf encoding the SST instead of individual components? That will make it easier to extend this with a row sample (and likely a table size) in the future.

Reply:
Fair point let me modify this.